package/MDAnalysis/analysis/msd.py

```diff
@@ -0,0 +1,421 @@
+unwrapped trajectories and maintaining a relatively small elapsed time between
+saved frames. Additionally, corrections for finite size effects are sometimes
+employed along with various means of estimating errors [Yeh2004]_ [Bulow2020]_.
+The reader is directed to the following review, which describes many of the
+common pitfalls [Maginn2019]_. There are other ways to compute
+self-diffusivity, such as from a Green-Kubo integral. At this point in time,
+these methods are beyond the scope of this module.
+
+
+Note also that computation of MSDs is highly memory intensive. If this is
+proving a problem, judicious use of the ``start``, ``stop``, ``step`` indexing is required.
+
+References
+----------
+
+.. [Maginn2019] Maginn, E. J., Messerly, R. A., Carlson, D. J.; Roe, D. R.,
+                Elliott, J. R. Best Practices for Computing Transport
+                Properties 1. Self-Diffusivity and Viscosity from Equilibrium
+                Molecular Dynamics [Article v1.0]. Living J. Comput. Mol. Sci.
+                2019, 1 (1).
+
+.. [Yeh2004] Yeh, I. C.; Hummer, G. System-Size Dependence of Diffusion
+                Coefficients and Viscosities from Molecular Dynamics
+                Simulations with Periodic Boundary Conditions.
+                J. Phys. Chem. B 2004, 108 (40), 15873–15879.
+
+.. [Bulow2020] von Bülow, S.; Bullerjahn, J. T.; Hummer, G. Systematic
+                Errors in Diffusion Coefficients from Long-Time Molecular
+                Dynamics Simulations at Constant Pressure. 2020.
+                arXiv:2003.09205 [Cond-Mat, Physics:Physics].
+
+
+
+Classes
+-------
+
+.. autoclass:: EinsteinMSD
+    :members:
+    :inherited-members:
+
+"""
+
+from __future__ import division, absolute_import
+
+import numpy as np
+import logging
+from six import raise_from
+from ..due import due, Doi
+from .base import AnalysisBase
+
+logger = logging.getLogger('MDAnalysis.analysis.msd')
+
+due.cite(Doi("10.21105/joss.00877"),
+         description="Mean Squared Displacements with tidynamics",
+         path="MDAnalysis.analysis.msd",
+         cite_module=True)
+due.cite(Doi("10.1051/sfn/201112010"),
+         description="FCA fast correlation algorithm",
+         path="MDAnalysis.analysis.msd",
+         cite_module=True)
+del Doi
+
+
+class EinsteinMSD(AnalysisBase):
+    r"""Class to calculate Mean Squared Displacement by the Einstein relation.
+
+    Attributes
+    ----------
+    dim_fac : int
+        Dimensionality :math:`d` of the MSD.
+    timeseries : :class:`numpy.ndarray`
+        The averaged MSD with respect to lag-time.
+    msd_per_particle : :class:`numpy.ndarray`
+        The MSD of each individual particle with respect to lag-time.
+    n_frames : int
+        Number of frames in trajectory.
+    n_particles : int
+        Number of particles MSD was calculated over.
+
+    """
+
+    def __init__(self, u, select=None, msd_type='xyz', fft=True, **kwargs):
```

Comment:
I feel 'name CA' would be confusing as a default? Trying to find the diffusivity of a  set of backbone atoms is from what I can tell rare.

Reply:
I agree with @hmacdope that CA does not make sense here. 

Can `u` also be a `AtomGroup`? If so, `select=None` should select everything so that one can easily use AtomGroups that already contain what's needed.

If `u` can only be a `Universe` then I am tempted to make `select` a required argument to force the user to say exactly what they want. Analyses classes are  free to implement any call signature that they like (see the [example for NewAnalysis](https://www.mdanalysis.org/docs/documentation_pages/analysis/base.html#MDAnalysis.analysis.base.AnalysisBase)). `NewAnalysis(atomgroup, **kwargs)` and `NewAnalysis(universe, select=None, **kwargs)` are common but `NewAnalysis(universe, select, **kwargs)` is allowed.

Reply:
As it currently stands can only be a `Universe`. I will change `select` to be required as suggested?

Reply:
`select` no longer a kwarg

Reply:
None of your code requires a Universe, you could just make it accept an `AtomGroup` by initialising with `u.universe.trajectory`. I think that's really it.

Reply:
Is it typical to have `select=None` select all the atoms instead of directly setting `select='all'` as the default?

Reply:
I honestly don't remember.... there's been something in the select vs selection discussion, hasn't it? In terms of "pythonic" behavior I always expect `None` for a kwarg to give me the default as opposed to the semantic "nothing". You're right, though, that in this case one could be clearer and set the default to `"all"`. (`u.select_atoms(None)` gives an empty AtomGroup so I don't think that letting `None` pass through is the right approach.)